### ISSUE TITLE
camera not pointing at object center 

### ISSUE THREAD

**Issue by cduguet** [kwea123/nerf_pl#95]
I have a 360deg inward scene of a backpack, with 24 pictures. The training with `--spheric` seems successful but when evaluating, the cameras seem to be pointing at the floor, not pointing to the whole object. 

This is how the input views look like (downresed for faster upload):
![a1467_621_valorous_pack_28l_blue_b-001](https://user-images.githubusercontent.com/1536515/130327947-93211758-7a2e-46aa-818c-8342b986f5cf.png)
![a1467_621_valorous_pack_28l_blue_b-008](https://user-images.githubusercontent.com/1536515/130327951-0f762c7b-7cd0-43f1-bb2e-e2f77797c0ef.png)

The colmap looks good as well: 
 
<img width="1440" alt="Screenshot 2021-08-21 at 18 09 41" src="https://user-images.githubusercontent.com/1536515/130328042-b649c5bb-342f-4f70-ab39-c0854aebf33f.png">
Training was done using **dev** branch and with: 

```
python train.py    --dataset_name llff    --root_dir /host/home/ubuntu/datasets/backpack/colmap_nerf_pl/    --N_importance 64 --img_wh 300 300    --num_epochs 60 --
batch_size 1024    --optimizer adam --lr 5e-4    --lr_scheduler steplr --decay_step 10 20 --decay_gamma 0.5    --exp_name backpack --spheric
```
and eval:
```
python eval.py    --root_dir /host/home/ubuntu/datasets/backpack/colmap_nerf_pl/    --dataset_name llff --scene_name backpack    --img_wh 300 300 --N_importance 6
4 --ckpt_path ckpts/backpackl/epoch\=59.ckpt  --spheric_poses   
```
with result:
![backpack](https://user-images.githubusercontent.com/1536515/130328229-5af8010b-a218-41ce-b2f0-c5a46834c4ee.gif)

Most probably it has something to do with the [calculation of the dataset center and the spheric poses](https://github.com/kwea123/nerf_pl/blob/2ca26add4f85218a97f6825e3eb0b9db19e7bab5/datasets/llff.py#L178), as discussed as well [in the original repo](https://github.com/bmild/nerf/issues/34).
I can only think of a workaround: modify [output of spheric poses in here](https://github.com/kwea123/nerf_pl/blob/2ca26add4f85218a97f6825e3eb0b9db19e7bab5/datasets/llff.py#L157) by adding a constant number to the y axis, but I don't know if that will require me to train again.
Anyway, it seems weird that the poses' calculation are off. 
I am also attaching poses  and colmap output. 
https://drive.google.com/file/d/17Yh-Etkee4NUNkcEmTacNkR9yUqhDtgk/view?usp=sharing

What could the reason? I don't see any points of the pointcloud out of place, in order to calculate a center around 0,0.

**Comment by kwea123** [comment]
Hi, you are right, the output spheric poses is the problem. As the code comment says, the output poses are pointing 36 degrees downwards, that's why the backpack is not at the center. A quick fix is to change the `-np.pi/5` to zero and generate again. Please let me know if this fixes your problem!

**Comment by cduguet** [comment]
Thank you! I tried with 
```
spheric_poses += [spheric_pose(th, 0, radius*1.5)]
```

and it seems like I need to train it again because everything above the original viewpoint looks like: 

![backpack-0deg](https://user-images.githubusercontent.com/1536515/131574657-f6c81799-2e4b-44ba-a91c-5e2958bebe23.gif)

I'm trying to understand why is the original calculation fundamentally wrong, and what could be a better fix for it.

**Comment by cduguet** [comment]
Trained the network completely with the new poses calculation, and got this:

![backpack-0deg (1)](https://user-images.githubusercontent.com/1536515/131804514-269f8b8e-898e-46c8-acf8-38187d55a81d.gif)

Now I'm out of ideas. Any suggestion?

**Comment by kwea123** [comment]
Can you share the tensorboard logs for the 2 trainings?

**Comment by cduguet** [comment]
[Here's are the colmap files.](https://drive.google.com/file/d/17Yh-Etkee4NUNkcEmTacNkR9yUqhDtgk/view?usp=sharing)

[Here are the logs and the checkpoint used for the first experiment (not modifying poses). ](https://drive.google.com/file/d/1BAIKfTjI2z_v9hr5EXbKQli0iIpblviB/view?usp=sharing)

[Here are the logs and checkpoint for the second experiment (modified poses).](https://drive.google.com/file/d/1xIfRp01uGC6GFJOjEadkfx-agaE5tFEt/view?usp=sharing)

Thank you very much!! Let me know if there is a problem getting access to them.

**Comment by cduguet** [comment]
I managed to make it work with [yenchenlin/nerf-pytorch](https://github.com/yenchenlin/nerf-pytorch) using the same input images/camera poses. I will try looking for the difference in your code in order to find the bug. 

https://user-images.githubusercontent.com/1536515/133881382-59d0292d-1350-46cb-8729-92eb6ce94d39.mp4

**Comment by msd-mohit** [comment]
> I managed to make it work with [yenchenlin/nerf-pytorch](https://github.com/yenchenlin/nerf-pytorch) using the same input images/camera poses. I will try looking for the difference in your code in order to find the bug.
> 
>  backpack_spiral_050000_rgb.mp4

Could you share the config file that you made for yenchenlin/nerf-pytorch implementation?

**Comment by ricshaw** [comment]
Did you manage to solve this? I am encountering the same issue.
